### DISCUSSION TITLE
OTA for ESP32

### DISCUSSION THREAD

**Discussion by SlimRG** [nanoframework/Home#1368] [Ideas]
### Description

I see no samples of OTA, except Azuze.
I want to have local isolated system and have possibility to update esp32 by Bluetooth or FTP server.
How can I do it?

### How to solve the problem

Add samples and API.

```[tasklist]
### Tasks
```

**Comment by josesimoes** [comment]
This is because this feature is not available. Yet. 
Work has been started, but lack of free time and funding put it into an halt.

**Comment by SlimRG** [comment]
So... What's new about OTA?

**Comment by Ellerbach** [comment]
> What's new about OTA?

@SlimRG Still same situation. Now, if you are a commercial user and really want this feature, sponsoring it will drastically speed up its development! Contact @josesimoes if you are willing to sponsor this.

**Comment by Najsr** [comment]
Hello, any update on this topic?

**Reply by josesimoes** [reply]
Still on the backlog. Not enough bandwidth to start the work. Paying projects upfront...

**Reply by Najsr** [reply]
Is the board with all the to-dos public? Would atleast love to see what is coming next

**Comment by mackyb27** [comment]
I have implemented OTA with plugin support, configuration support and non aws connections and Will post it when it is complete. Ota approach requires lots of RAM so it Will probably only work on esp32 with psram, s2 or s3 i suppose. On c3 you can only load 4 plugins and some system libraries. Then there is an issue of system libraries and keeping track of them...  I have implemented eventbus and separate plugin storage as well as system libraries repository. However one of the biggest issue with such approach is that youre stuck with certain version of firmware and nanoframework libraries. However i still think this is THE feature that could elevate nanoframework to New hights.

**Reply by josesimoes** [reply]
Nice! This is getting close to the top on my todo list.
Curious on how you've implemented it.

**Reply by SlimRG** [reply]
any updates?

**Comment by renerlemes** [comment]
@mackyb27 any updates?

**Reply by networkfusion** [reply]
Not yet

**Comment by SlimRG** [comment]
any updates?

**Comment by SunNull** [comment]
any updates?

**Comment by josesimoes** [comment]
Getting there!
